doc/working_example.md

```diff
@@ -0,0 +1,825 @@
+# Genalyzer Through Examples
+## A Minimal Working Example
+The workflow of Genalyzer is best understood with the help of a working example. We use Python to illustrate this workflow in the current section. The data that we will process using Genalyzer for this example is also generated using Genalyzer. It consists of a ``300 KHz`` tone sampled at ``3 MSPS``. The only source of noise in this synthetic data is quantization noise.
+<br/><br/>
+
+### Compute FFT
+We begin with the initial steps involved, _i.e.,_, configuring Genalyzer and performing FFT on the generated data. Since spectral analysis is the core of Genalyzer, computing FFT is the first important step. Expand the code-snippet below for details.
+<details>
+  <summary>Code Snippet</summary>
+
+``` Python
+import genalyzer.advanced as gn
+
+#
+# Signal 
+#
+npts = 30000 # number of points in the signal
+freq = 300000 # tone frequency
+phase = 0.0 # tone phase
+ampl_dbfs = -1.0 # amplitude of the tone in dBFS
+qnoise_dbfs = -63.0  # quantizer noise in dBFS
+fsr = 2.0 # full-scale range of I/Q components of the complex tone
+ampl = (fsr / 2) * 10 ** (ampl_dbfs / 20) # amplitude of the tone in linear scale
+qnoise = 10 ** (qnoise_dbfs / 20) # quantizer noise in linear scale
+qres = 12  # data resolution
+code_fmt = gn.CodeFormat.TWOS_COMPLEMENT # integer data format
+fs = 3e6 # sample-rate of the data
+
+#
+# FFT configuration
+#
+navg = 1 # number of FFT averages
+nfft = int(npts/navg) # FFT-order
+window = gn.Window.NO_WINDOW # window function to apply
+
+#
+# Genarate signal for analysis
+#
+awfi = gn.cos(npts, fs, ampl, freq, phase)
+awfq = gn.sin(npts, fs, ampl, freq, phase)
+qwfi = gn.quantize(awfi, fsr, qres, qnoise, code_fmt)
+qwfq = gn.quantize(awfq, fsr, qres, qnoise, code_fmt)
+
+#
+# Compute FFT
+#
+fft_cplx = gn.fft(qwfi, qwfq, qres, navg, nfft, window, code_fmt)
+```
+</details>
+
+We will expand this code snippet to illustrate the various capabilities of Genalyzer. We postpone the discussion on how to generate synthetic data using Genalyzer and the related knobs available. We begin first with the FFT-computing step in the above snippet.
+
+The arguments that ``fft()`` takes are shown on [here](#genalyzer.advanced.advanced.fft). As shown on this page, it takes a variable number of arguments to handle several usecases. The usecases are split based on how the I/Q samples are represented, _i.e._, whether the samples are complex-valued, or interleaved I/Q, or split into I and Q separately. These categories are further split based on whether the samples are represented as floating-point or fixed-point integer data. In the latter case, the resolution in bits and code-format are the required inputs. The supported datatypes for the input data vector are: ``complex128``, ``float64``, ``int16``, ``int32``, ``int64``. 
```

Comment:
Refrain from using "i.e." a lot. It can be distracting.

I don't really understand the point of this paragraph. Are you trying to explain how to use the fft function? its flexibility? or its nuanced API? 

For individual arguments just refer users to the reference doc. This shouldn't be a copy+paste of that here. 

This section is "Compute FFT" but it probably needs to be more specific on the data types used since that's what drives everything.